src/t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.cxx

```diff
@@ -0,0 +1,175 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <t8_geometry/t8_geometry_base.hxx>
+#include <t8_cmesh/t8_cmesh_types.h>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.hxx>
+#include <t8_geometry/t8_geometry_implementations/t8_geometry_lagrange.h>
+#include <t8_eclass.h>
+
+t8_geometry_lagrange::t8_geometry_lagrange (int dim): t8_geometry_with_vertices (dim, "")
+{
+  T8_ASSERT (0 <= dim && dim <= 3);
+  size_t num_chars = 100;
+  char *name_tmp = T8_ALLOC (char, num_chars);
+
+  snprintf (name_tmp, num_chars, "t8_geom_lagrange_%i", dim);
+  name = name_tmp;
+  dimension = dim;
+}
+
+t8_geometry_lagrange::~t8_geometry_lagrange ()
+{
+  T8_FREE ((char *) name);
+}
+
+/**
+ * Finite element mapping.
+ * For linear elements, it gives the same result as
+ * t8_geom_compute_linear_geometry (active_tree_class, active_tree_vertices, ref_coords, num_coords, out_coords);
+ */
+void
+t8_geometry_lagrange::t8_geom_evaluate (t8_cmesh_t cmesh, t8_gloidx_t gtreeid, const double *ref_coords,
+                                        const size_t num_coords, double *out_coords) const
+{
+  if (num_coords != 1)
+    SC_ABORT ("Error: Batch computation of geometry not yet supported.");
```

Comment:
It would bring a big performance boost if you implement this. We are also implementing this also for all other geometries.